app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java

```diff
@@ -827,4 +831,88 @@ public void setExampleAppsInConfig(MongoTemplate mongoTemplate) {
+        final Config config = mongoTemplate.findOne(
+                configQuery,
+                Config.class
+        );
+
+        if (config == null) {
+            // No template organization configured. Nothing to migrate.
+            return;
+        }
+
+        final String organizationId = config.getConfig().getAsString("organizationId");
+
+        final org.springframework.data.mongodb.core.query.Query query = query(where(FieldName.ORGANIZATION_ID).is(organizationId));
+        query.fields().include("_id");
+
+        // Get IDs of applications in the template org.
+        final List<String> applicationIds = mongoTemplate
+                .find(query, Application.class)
+                .stream()
+                .map(BaseDomain::getId)
+                .collect(Collectors.toList());
+
+        // Get IDs of actions in the template org.
+        final List<String> actionIds = mongoTemplate
+                .find(query, Action.class)
+                .stream()
+                .map(BaseDomain::getId)
+                .collect(Collectors.toList());
+
+        // Get pages that are not in applications in the template org, and have template org's action IDs in their
+        // layoutOnload lists.
+        final Criteria incorrectActionIdCriteria = new Criteria().orOperator(
+                where("layouts.layoutOnLoadActions").elemMatch(new Criteria().elemMatch(where("_id").in(actionIds))),
+                where("layouts.publishedLayoutOnLoadActions").elemMatch(new Criteria().elemMatch(where("_id").in(actionIds)))
+        );
+        final List<Page> pagesToFix = mongoTemplate.find(
+                query(where(FieldName.APPLICATION_ID).not().in(applicationIds))
+                        .addCriteria(incorrectActionIdCriteria),
+                Page.class
+        );
+
+
+        for (Page page : pagesToFix) {
+            for (Layout layout : page.getLayouts()) {
+                final ArrayList<HashSet<DslActionDTO>> layoutOnLoadActions = new ArrayList<>();
+                if (layout.getLayoutOnLoadActions() != null) {
+                    layoutOnLoadActions.addAll(layout.getLayoutOnLoadActions());
+                }
+                if (layout.getPublishedLayoutOnLoadActions() != null) {
+                    layoutOnLoadActions.addAll(layout.getPublishedLayoutOnLoadActions());
+                }
+                for (HashSet<DslActionDTO> actionSet : layoutOnLoadActions) {
+                    for (DslActionDTO actionDTO : actionSet) {
+                        final String actionName = actionDTO.getName();
+                        final Action action = mongoTemplate.findOne(
+                                query(where(FieldName.PAGE_ID).is(page.getId()))
+                                        .addCriteria(where(FieldName.NAME).is(actionName)),
+                                Action.class
+                        );
+                        if (action != null) {
+                            actionDTO.setId(action.getId());
```

Comment:
Can we add a minor comment here?
```suggestion
                           // update the erroneous action id (template action id) to the cloned action id
                            actionDTO.setId(action.getId());
```